### PR TITLE
[KARAF-6431] - Maven plugin feature to define custom bundle location prefixes

### PR DESCRIPTION
**Pull request by ricardosilva88**
Continuation of https://github.com/apache/karaf/pull/879, but this time aiming to merge into master

### CODE REVIEW

**Comment by jbonofre** [comment]
Your change is breaking the `minimal` distribution build. I will take a look.

**Comment by ricardosilva88** [comment]
I saw, [ERROR] Failed to execute goal org.apache.karaf.tooling:karaf-maven-plugin:4.3.0-SNAPSHOT:assembly (process-resources) on project apache-karaf-minimal: Unable to build assembly: NullPointerException, I will have a look also, don't see any direct relation between my code and this error, but will look into it.

**Comment by ricardosilva88** [comment]
@jbonofre, I just fixed the NPE, it was missing the following line - https://github.com/apache/karaf/pull/882/files#diff-90746339f4ad3d5ce78192a35cbd86c8R824

**Comment by ricardosilva88** [comment]
@jbonofre , can you please review my changes? ty

**Comment by ricardosilva88** [comment]
@jbonofre any update on this PR? Is missing something from my side?

**Comment by jbonofre** [comment]
retest this please

**Comment by jbonofre** [comment]
retest this please

**Comment by jbonofre** [comment]
retest this please

**Comment by ricardosilva88** [comment]
"ERROR: Maven Home /home/jenkins/tools/maven/apache-maven-3.6.0 doesnt exist"

**Comment by jbonofre** [comment]
retest this please

**Comment by jbonofre** [comment]
retest this please

**Comment by jbonofre** [comment]
H41 Jenkins executor doesn't have any Maven installation. I'm dealing with infra.

**Comment by jbonofre** [comment]
In the meantime, I will fully test the PR locally to merge it.

**Comment by jbonofre** [comment]
retest this please

File: features/core/src/test/java/org/apache/karaf/features/internal/download/impl/DownloadManagerHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package org.apache.karaf.features.internal.download.impl;
```
Comment (jbonofre) [inline features/core/src/test/java/org/apache/karaf/features/internal/download/impl/DownloadManagerHelperTest.java]: This file doesn't contain the Apache license header, can you please add it ?

Comment (ricardosilva88) [reply features/core/src/test/java/org/apache/karaf/features/internal/download/impl/DownloadManagerHelperTest.java]: sure, done

**Comment by jbonofre** [comment]
retest this please

**Pull request merged**